tests/routes/native/v1/offers_test.py

```diff
@@ -177,34 +176,21 @@ def test_get_thing_offer(self, app):
         assert not response.json["isExpired"]
         assert response.json["canExpire"] is True
 
-    def test_get_digital_offer_book_cannot_exire(self, app):
-        product = ProductFactory(thumbCount=1)
-        offer_type = ThingType.LIVRE_AUDIO
-        offer = OfferFactory(type=str(offer_type), product=product)
-        ThingStockFactory(offer=offer, price=12.34)
-
-        offer_id = offer.id
-        with assert_num_queries(1):
+    def test_get_digital_offer_with_available_activation_and_no_expiration_date(self, app):
+        stock = StockWithActivationCodesFactory()
+        offer_id = stock.offer.id
+        # QUERY COUNT
+        # 1: the offer
+        # 2: 1 query to get available_activation_code for each offer.stocks
+        with assert_num_queries(2):
```

Comment:
(détail / style)

C'est du code que tu as recopié et dont tu n'es pas l'auteur, mais je commente quand même.

Je trouve peu pratique de numéroter les requêtes. Si une requête SQL s'ajoute un jour, on devra tout renuméroter et ça fait un diff un peu verbeux. Exemple : https://github.com/pass-culture/pass-culture-api/commit/f03bf3b854175b3d3196120a6d8b3bfa83262656#diff-cd60213fa8a4b9a6575aec29380da7aa701dc5e72775d85ec58f0a67ba45346bL348-R349

Je trouve donc la formulation suivante plus facile à mettre à jour :
```suggestion
        queries = 1  # select offer
        queries += 1   # get available_activation_code for each offer.stocks
        with assert_num_queries(queries):
```

Reply:
C'est une bonne idée.

Reply:
Done

Reply:
En ce qui concerne ta remarque sur dateExpiration nulle, on a vu le code avec @antoinewg et je pense que côté app native c'est ok.
Je laisse @antoinewg confirmer.